### PR TITLE
DOCSP-41557: New v4.7 commands and methods

### PR DESCRIPTION
**Pull request by norareidy**
JIRA - https://jira.mongodb.org/browse/DOCSP-41557
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/laravel/DOCSP-41557-stage/fundamentals/database-collection/

### Checklist

- [ ] Add tests and ensure they pass
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Update documentation for new features

### CODE REVIEW

File: docs/fundamentals/database-collection.txt [hunk 1]
```diff
@@ -161,17 +161,105 @@ as in the preceding example, but the query is constructed by using the
 List Collections
 ----------------
 
-To see information about each of the collections in a database, call the 
-``listCollections()`` method.
+This section describes how to see information about each collection in a database
+in the following ways:
```
Comment (rustagir) [inline docs/fundamentals/database-collection.txt]:
```suggestion
You can take either of the following actions to see information
about each collection in a database:
```

File: docs/fundamentals/database-collection.txt [hunk 2]
```diff
@@ -161,17 +161,105 @@ as in the preceding example, but the query is constructed by using the
...
+- :ref:`laravel-list-coll-command`
+- :ref:`laravel-list-coll-methods`
+
+.. _laravel-list-coll-command:
+
+Run a Shell Command
+~~~~~~~~~~~~~~~~~~~
+
+You can list the collections in a database by running the following shell
+command from your project root:
```
Comment (rustagir) [inline docs/fundamentals/database-collection.txt]:
```suggestion
You can list the collections in a database by running the following
command in your shell from the root directory of your project:
```

File: docs/fundamentals/database-collection.txt [hunk 3]
```diff
@@ -161,17 +161,105 @@ as in the preceding example, but the query is constructed by using the
...
+
+.. code-block:: bash
+
+   php artisan db:show
+
+This command outputs information about the configured database and lists its
+collections under the ``Table`` header. For more information about the ``db:show``
+command, see `Laravel: New DB Commands <https://blog.laravel.com/laravel-new-db-commands-and-more>`__.
```
Comment (rustagir) [inline docs/fundamentals/database-collection.txt]:
```suggestion
This command outputs information about the configured database and lists the
collections under the ``Table`` header. For more information about the ``db:show()``
command, see `Laravel: New DB Commands <https://blog.laravel.com/laravel-new-db-commands-and-more>`__
on the official Laravel blog.
```

File: docs/fundamentals/database-collection.txt [hunk 4]
```diff
@@ -161,17 +161,105 @@ as in the preceding example, but the query is constructed by using the
...
 
 .. code-block:: php
 
    $collections = DB::connection('mongodb')->getMongoDB()->listCollections();
 
+List Collection Fields
+----------------------
+
+This section describes how to see information about each field in a collection
+in the following ways:
```
Comment (rustagir) [inline docs/fundamentals/database-collection.txt]:
```suggestion
You can take either of the following actions to see information
about each field in a collection:
```

File: docs/fundamentals/database-collection.txt [hunk 5]
```diff
@@ -161,17 +161,105 @@ as in the preceding example, but the query is constructed by using the
...
+- :ref:`laravel-list-fields-command`
+- :ref:`laravel-list-fields-methods`
+
+.. _laravel-list-fields-command:
+
+Run a Shell Command
+~~~~~~~~~~~~~~~~~~~
+
+You can see a list of fields in a collection by running the following shell
+command from your project root:
```
Comment (rustagir) [inline docs/fundamentals/database-collection.txt]:
```suggestion
You can see a list of fields in a collection by running the following
command in your shell from the root directory of your project:
```

File: docs/fundamentals/database-collection.txt (hunk 3 above)
Comment (rustagir) [reply docs/fundamentals/database-collection.txt]: applies to the other usage as well

File: docs/fundamentals/database-collection.txt [hunk 6]
```diff
@@ -161,17 +161,105 @@ as in the preceding example, but the query is constructed by using the
...
+.. _laravel-list-fields-methods:
+
+Call Schema Methods
+~~~~~~~~~~~~~~~~~~~
+
+You can list the fields in a collection by calling the ``Schema::getColumns()``
+schema builder method in your application.
+
+You can also call the following methods to return more information about the
+collection fields:
```
Comment (rustagir) [inline docs/fundamentals/database-collection.txt]:
```suggestion
You can also use the following methods to return more information about the
collection fields:
```

File: docs/fundamentals/database-collection.txt [hunk 7]
```diff
@@ -161,17 +161,105 @@ as in the preceding example, but the query is constructed by using the
...
+
+- ``Schema::hasColumn(string $field)``: checks if the specified field exists
+  in at least one document
+- ``Schema::hasColumns(string[] $fields)``: checks if each specified field exists
+  in at least one document
```
Comment (rustagir) [inline docs/fundamentals/database-collection.txt]:
```suggestion
- ``Schema::hasColumn(string $<field name>)``: checks if the specified field exists
  in at least one document
- ``Schema::hasColumns(string[] $<field names>)``: checks if each specified field exists
  in at least one document
```

**Review by rustagir — APPROVED** [review]
approving with some copy change suggestions!

File: docs/fundamentals/database-collection.txt [hunk 8]
```diff
@@ -161,17 +161,107 @@ as in the preceding example, but the query is constructed by using the
...
+You can list the fields in a collection by calling the ``Schema::getColumns()``
+schema builder method in your application.
+
+You can also use the following methods to return more information about the
+collection fields:
+
+- ``Schema::hasColumn(string $<field name>)``: checks if the specified field exists
+  in at least one document
+- ``Schema::hasColumns(string[] $<field names>)``: checks if each specified field exists
+  in at least one document
```
Comment (GromNaN) [inline docs/fundamentals/database-collection.txt]: This need to mention that MongoDB is a schemaless database, so this functions checks the data, not the schema like in traditional databases. If the collection doesn't exist or is empty, this function always return `false`.

Comment (norareidy) [reply docs/fundamentals/database-collection.txt]: I added a note clarifying this, let me know what you think

**Review by GromNaN — APPROVED** [review]

**Pull request merged**